components/brave_rewards/resources/rewards_panel/components/panel.tsx

```diff
@@ -84,11 +133,15 @@ export function Panel () {
   }
 
   return (
-    <div>
-      <div className='rewards-panel' data-test-id='rewards-panel'>
-        {userType !== 'unconnected' ? renderFull() : <LimitedView />}
-      </div>
-      <PanelOverlays />
-    </div>
+    onboardingResult || !rewardsEnabled || needsCountry
+      ? <div className='rewards-panel' data-test-id='rewards-panel'>
```

Comment:
I wonder if we should move this into `app.tsx` to avoid duplication here, so that in `app.tsx` we'd have something like:

```tsx
<style.root>
  {loading ? <Loading /> :
       <div className='rewards-panel' data-test-id='rewards-panel'>
         <Panel key={panelKey} />
       </div>
  }
</style.root>
```

Reply:
The way the code is now those attributes don't extend to the overlays. If we change it this way they would. Although, I am not sure it matters. I don't see where they actually get used.

Reply:
Updated in https://github.com/brave/brave-core/commit/1bb3417cf7a2cf42118a77eddba9630547cb4692